src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/TemplateCollectionFunctionalTests.cs

```diff
@@ -303,6 +329,32 @@ public async Task GivenSameInputData_WithDifferentTemplateSource_WhenConvert_Res
             Assert.Equal(imageResult, folderResult);
         }
 
+        [Theory]
+        [MemberData(nameof(GetCdaDataAndTemplateSources))]
+        public async Task GivenCdaSameInputData_WithDifferentTemplateSource_WhenConvert_ResultShouldBeIdentical(string inputFile, string defaultTemplateDirectory, string rootTemplate)
+        {
+            var folderTemplateProvider = new CdaTemplateProvider(defaultTemplateDirectory);
+
+            var templateProviderFactory = new TemplateCollectionProviderFactory(new MemoryCache(new MemoryCacheOptions()), Options.Create(new TemplateCollectionConfiguration()));
+            var templateProvider = templateProviderFactory.CreateTemplateCollectionProvider(ImageInfo.GetDefaultTemplateImageReferenceByDatatype(DataType.Cda), string.Empty);
+            var imageTemplateProvider = new CdaTemplateProvider(await templateProvider.GetTemplateCollectionAsync(CancellationToken.None));
+
+            var cdaProcessor = new CdaProcessor();
+            var inputContent = File.ReadAllText(inputFile);
+
+            var imageResult = cdaProcessor.Convert(inputContent, rootTemplate, imageTemplateProvider);
+            var folderResult = cdaProcessor.Convert(inputContent, rootTemplate, folderTemplateProvider);
+
+            var imageResultObject = JObject.Parse(imageResult);
+            var folderResultObject = JObject.Parse(folderResult);
+
+            // Remove DocumentReference, where date is different every time conversion is run and gzip result is OS dependent
+            imageResultObject["entry"]?.Last()?.Remove();
+            folderResultObject["entry"]?.Last()?.Remove();
```

Comment:
Just curious, will date for hl7v2's conversion result be different every time? And will OS effect hl7v2?

Reply:
Hl7v2 result is the same, while C-CDA result will be different every time because the conversion time is attached.